di/src/main/java/com/interface21/beans/factory/support/DefaultListableBeanFactory.java

```diff
@@ -1,78 +1,87 @@
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
-import com.interface21.beans.BeanInstantiationException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.interface21.beans.BeanUtils;
 import com.interface21.beans.factory.BeanFactory;
-import com.interface21.context.stereotype.Component;
+import com.interface21.beans.factory.config.BeanDefinition;
 
-public class DefaultListableBeanFactory implements BeanFactory {
+public class DefaultListableBeanFactory implements BeanFactory, BeanDefinitionRegistry {
 
     private static final Logger log = LoggerFactory.getLogger(DefaultListableBeanFactory.class);
 
-    private final BeanDefinitionReader beanDefinitionReader;
+    private final Map<Class<?>, BeanDefinition> beanDefinitionMap = new ConcurrentHashMap<>();
     private final BeanRegistry beanRegistry;
 
-    public DefaultListableBeanFactory(String[] basePackages) {
-        this.beanDefinitionReader = new AnnotationBeanDefinitionReader(basePackages);
+    public DefaultListableBeanFactory() {
         this.beanRegistry = new DefaultBeanRegistry();
     }
 
     public void initialize() {
-        beanDefinitionReader.loadBeanDefinitions(new Class[] {Component.class});
-        beanDefinitionReader
-                .getBeanDefinitions()
+        beanDefinitionMap
+                .values()
                 .forEach(
                         beanDefinition -> {
                             Class<?> beanClazz = beanDefinition.getType();
-                            beanRegistry.registerBean(getBeanOrCreate(beanClazz));
+                            beanRegistry.registeredBean(getBean(beanClazz));
                         });
     }
 
+    @Override
+    public void registerBeanDefinition(Class<?> clazz, BeanDefinition beanDefinition) {
+        beanDefinitionMap.put(clazz, beanDefinition);
+    }
+
     @Override
     public Set<Class<?>> getBeanClasses() {
-        return beanDefinitionReader.getBeanClasses();
+        return beanDefinitionMap.keySet();
     }
 
     @Override
-    public <T> T getBean(final Class<T> clazz) {
-        return (T) beanRegistry.getBean(clazz);
+    public List<BeanDefinition> getBeanDefinitions() {
+        return List.copyOf(beanDefinitionMap.values());
     }
 
     @Override
-    public Object getBeanOrCreate(Class<?> clazz) {
-        Class<?> concreteClazz =
-                BeanFactoryUtils.findConcreteClass(clazz, getBeanClasses())
-                        .orElseThrow(() -> new BeanClassNotFoundException(clazz.getSimpleName()));
+    public int getBeanDefinitionCount() {
+        return getBeanDefinitions().size();
+    }
 
-        if (beanRegistry.registeredBean(concreteClazz)) {
-            return beanRegistry.getBean(concreteClazz);
+    @Override
+    public <T> T getBean(final Class<T> clazz) {
+        Object bean = beanRegistry.getBean(clazz);
+        if (bean != null) {
+            return (T) bean;
```

Comment:
```suggestion
            return clazz.cast(bean);
```

이런식으로 표현한다면 warning도 사라질 것 같네요 👍